BeeSwift/GoalViewController.swift

```diff
@@ -353,7 +353,10 @@ class GoalViewController: UIViewController,  UIScrollViewDelegate, DatapointTabl
     }
 
     @objc func goalImageTapped() {
-        self.goalImageScrollView.setZoomScale(self.goalImageScrollView.zoomScale == 1.0 ? 2.0 : 1.0, animated: true)
+        let possibleNextStepUp = self.goalImageScrollView.zoomScale + 2.0
+        let nextZoomLevel = possibleNextStepUp <= self.goalImageScrollView.maximumZoomScale ? possibleNextStepUp : self.goalImageScrollView.minimumZoomScale
+        
+        self.goalImageScrollView.setZoomScale(nextZoomLevel, animated: true)
```

Comment:
This seems reasonable for iPhone. I'm less sure it makes sense for iPad though? Perhaps the max zoom level should be a function of the screen size?

Reply:
I still found it quite OK to double tap to zoom on the iPad. The bigger issue for me was that the graph image being used quickly becomes blurry since we are using the lower resolution png rather than the svg. The user can still pinch-to-zoom and can do so at the previous level as well as now also at other levels.